airflow/models/dagrun.py

```diff
@@ -334,6 +334,34 @@ def find(
 
         return qry.order_by(DR.execution_date).all()
 
+    @staticmethod
+    @provide_session
+    def find_duplicate(
+        dag_id: str,
+        run_id: str,
+        execution_date: datetime,
+        session: Session = None,
+    ) -> Optional['DagRun']:
+        """
+        Returns a dag run if there is an existing run for a dag at a specific run_id and execution_date.
```

Comment:
```suggestion
        Return an existing run for the DAG with a specific run_id or execution_date.
        
        *None* is returned if no such DAG run is found.
```

I wonder if this function should be on `DAG` instead, since this needs a DAG ID.

If we are going to put this on `DagRun`, this should be a classmethod (so you can do `session.query(cls)` below).

Reply:
I think it's fine in `DagRun` since the function itself is related to finding DAG runs and has some similarities with `find` function

Let me know your thoughts.